lib/search.rb

```diff
@@ -7,6 +7,16 @@ class Search
   cattr_accessor :preloaded_topic_custom_fields
   self.preloaded_topic_custom_fields = Set.new
 
+  def self.on_preload(&blk)
+    (@preload ||= Set.new) << blk
+  end
+
+  def self.preload(search, object)
+    if @preload
+      @preload.each { |preload| preload.call(search, object) }
```

Comment:
Perhaps: 
```suggestion
  def self.preload(results, search)
    if @preload
      @preload.each { |preload| preload.call(results, search) }
```
to match the arguments in the call (line 295) - the `results` object and a `Search` instance?

Reply:
Agreed, made the change.